VectorSearchAiAssistant.SemanticKernel/Plugins/Core/TextSummaryPlugin.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI;
+using Microsoft.SemanticKernel.Orchestration;
+using System.ComponentModel;
+
+namespace VectorSearchAiAssistant.SemanticKernel.Plugins.Core
+{
+    public class TextSummaryPlugin
+    {
+        private readonly ISKFunction _summarizeConversation;
+        private readonly IKernel _kernel;
+
+        public TextSummaryPlugin(
+            string promptTemplate,
+            int maxTokens,
+            IKernel kernel)
+        {
+            _kernel = kernel;
+            _summarizeConversation = kernel.CreateSemanticFunction(
+                promptTemplate,
+                pluginName: nameof(TextSummaryPlugin),
+                description: "Given a text, summarize the text.",
+                requestSettings: new OpenAIRequestSettings
+                {
+                    MaxTokens = maxTokens,
+                    Temperature = 0.1,
+                    TopP = 0.5
```

Comment:
Did you test these settings? In other configurations I had this as determinstic as possible with temp=0 and top_p=1 to keep it from going off the rails. Not sure if these are ok. just haven't tried these values.

Reply:
For summarization, I prefer a little bit of freedom, but will add a comment to point out that folks should try with temp=0 and top_p=1 as well.